server/controllers/articles.js

```diff
@@ -317,5 +320,206 @@ class Articles {
       errorHandler.errorResponse(res, error);
     }
   }
+
+  /**
+   * @static
+   * @description convert mutliple tags separeted by space or commas into an array of tags
+   * @param {string} tag
+   * @returns {string[]} tagList
+  */
+  static tagsToArray(tag) {
+    const tagList = tag.split(/[ ,]+/)
+      .filter(Boolean)
+      .map(item => item.replace(/,+/g, ''));
+    return tagList;
+  }
+
+  /**
+   * search filter for articles
+   * @author Mutombo jean-vincent
+   * @static
+   * @param {*} req
+   * @param {*} res
+   * @returns {object} response
+   * @memberof Articles
+  */
+  static async search(req, res) {
+    // get query string
+    const {
+      author, title, tag, keyword, offset = 0, limit = 20
+    } = req.query;
+
+    const pagination = { offset, limit };
+
+    try {
+      if (author) {
+        Articles.filterByAuthor(res, author, pagination);
+      } else if (title) {
+        Articles.filterByTitle(res, title, pagination);
+      } else if (keyword) {
```

Comment:
I think keyword and tag is just one thing
![image](https://user-images.githubusercontent.com/27511264/57270605-57116180-708c-11e9-9282-d271f0f73ca6.png)

Reply:
> I think keyword and tag is just one thing
> ![image](https://user-images.githubusercontent.com/27511264/57270605-57116180-708c-11e9-9282-d271f0f73ca6.png)

Hey @dusmel. Thank you for the feedback. When I have seen the term `keyword`, I personally understood that `it can be any word or sentence inside the article or which is related to the article.` So I am actually searching the keyword in the article title, body or article description. I should also add the case where we can search the keyword inside the tag.